### PR TITLE
Marabunta first version as "setup"

### PR DESCRIPTION
**Pull request by oleksandrpaziuk**
Change the first version name from `x.y.z` to `setup`
Change versioning system from `distutils.StrictVersion` to Marabunta `MarabuntaVersion`
use only 3 and 5 digits version numbers (ex.:10.0.1 or 10.0.1.0.1)

### CODE REVIEW

File: marabunta/version.py [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+
+import re
+from distutils.version import Version
+
+
+FIRST_VERSION = 'setup'
+
+
+class MarabuntaVersion(Version):
+    """Version numbering for anal retentives and software idealists.
```
Comment (simahawk) [inline marabunta/version.py]: :open_mouth: :red_circle: :hand:  :stop_sign: :fire:

File: tests/test_version.py [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+        ('setup', '0.0.0', -1),
+        ('0.0.0.0.0', 'setup', 1),
+        ('setup', '0.0.0.0.0', -1),
+        ('1.2.3', 'setup', 1),
+        ('setup', '4.5.6', -1),
+        ('11.0.1.0.0', 'setup', 1),
+        ('setup', '10.0.0.0.1', -1),
+    )
+
+    for v1, v2, expected in versions:
```
Comment (simahawk) [inline tests/test_version.py]: this is a bit weird...
I'd split the test in 3: test good versions, test bad versions, test comparison.
Test comparison should not call _cmp but do actual comparisons like `self.assertGreater(v1, v2)`.

File: marabunta/version.py [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+
+    def parse(self, version_str):
+        match = self.version_re.match(version_str)
+        if not match:
+            raise ValueError("invalid version number '%s'" % version_str)
+
+        if match.string == FIRST_VERSION:
+            self.version = match.string
+        else:
+            (major, minor, patch, revision, build) = \
```
Comment (simahawk) [inline marabunta/version.py]: not mandatory, but you can name groups  to ease access

File: marabunta/model.py [hunk 4]
```diff
@@ -14,6 +16,52 @@
...
+
+    def parse(self, vstring):
+        match = self.version_re.match(vstring)
+        if not match:
+            raise ValueError, "invalid version number '%s'" % vstring
+
+        if match.string == FIRST_VERSION:
+            self.version = match.string
+        else:
+            (major, minor, patch, prerelease, prerelease_num) = \
```
[the remark this answers is not in the document]
Comment (oleksandrpaziuk) [reply marabunta/model.py]: I don't know how you write a review for the amended file but now MarabuntaVersion inherits from `distutils.version.Version`
Version haven't any `parse` method
https://github.com/python/cpython/blob/3.5/Lib/distutils/version.py#L31

File: marabunta/version.py (hunk 1 above)
Comment (oleksandrpaziuk) [reply marabunta/version.py]: https://github.com/python/cpython/blob/3.5/Lib/distutils/version.py#L95

File: tests/test_version.py (hunk 2 above)
Comment (oleksandrpaziuk) [reply tests/test_version.py]: I take tests manere from https://github.com/python/cpython/blob/3.5/Lib/distutils/tests/test_version.py#L18

Comment (simahawk) [reply tests/test_version.py]: as the bad wording in docstring: we don't have to copy everything... we can take examples and improve :wink: 
Not a blocker anyway.

File: marabunta/version.py (hunk 1 above)
Comment (oleksandrpaziuk) [reply marabunta/version.py]: Fixed

File: marabunta/version.py [hunk 5]
```diff
@@ -10,7 +10,8 @@
 
 
 class MarabuntaVersion(Version):
-    """Version numbering for anal retentives and software idealists.
+
```
Comment (leemannd) [inline marabunta/version.py]: This line can be removed

**Comment by simahawk** [comment]
squashed and merged. Tnx Olek!

**Comment by yvaucher** [comment]
It is very sad that we broke the natural string ordering :-/

**Comment by simahawk** [comment]
> It is very sad that we broke the natural string ordering :-/

let's talk about this :wink:

**Pull request merged**